src/mpid/ch4/netmod/ofi/ofi_impl.h

```diff
@@ -289,10 +289,6 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_OFI_mr_bind(struct fi_info *prov, struct fid_
     if (prov->domain_attr->mr_mode == FI_MR_ENDPOINT) {
         /* Bind the memory region to the endpoint */
         MPIDI_OFI_CALL(fi_mr_bind(mr, &ep->fid, 0ULL), mr_bind);
-        /* Bind the memory region to the counter */
-        if (cntr != NULL) {
-            MPIDI_OFI_CALL(fi_mr_bind(mr, &cntr->fid, 0ULL), mr_bind);
-        }
```

Comment:
According to https://ofiwg.github.io/libfabric/v1.15.0/man/fi_mr.3.html:
```
fi_mr_bind

The fi_mr_bind function associates a memory region with a counter or endpoint. Counter bindings are needed by providers that support the generation of completions based on fabric operations. Endpoint bindings are needed if the provider associates memory regions with endpoints (see FI_MR_ENDPOINT).

When binding with a counter, the type of events tracked against the memory region is based on the bitwise OR of the following flags.
```

If this is a CXI workaround, add a capability macro and use the `#ifdef`. If we don't need bind to counter, please support it with documentation and explanations.

Reply:
```
Since MPICH uses only local counters, only an fi_ep_bind call is
required for local counters. Previously, this fi_mr_bind call
was causing a crash during fi_mr_enable while performing RMA init
with the CXI provider.
```
Sorry for the delayed response. I was talking to the libfabric and cxi developers to confirm my understanding of the fi_mr_bind call for coutners. This is relevant to all FI_MR_ENDPOINT layers. 
I've added this in the commit. Do you mean documentation in the code explaining why the bind call is not needed?

Reply:
> Do you mean documentation in the code explaining why the bind call is not needed?

Just need to explain to me (and a random future maintainer). You may just add the explanation to the commit message.

Reply:
> Previously, this fi_mr_bind call was causing a crash during fi_mr_enable while performing RMA init with the CXI provider.

Was this a CXI bug?

Reply:
👍

Reply:
@nitbhat Regarding the first comment above, do you have any remote counter that needed to enable? In the OpenSHMEM side, I am still seeing the progress is stuck with no update through the counters. I have used `fi_mr_bind` with `FI_REMOTE_WRITE` flag to enable this. 

Thanks to @nusislam who pointed me to this PR.

Reply:
@wrrobin, No in MPICH, we only use local counters and do not use remote counters. For that reason, I removed this bind call. 

While debugging, at one stage, I did have `fi_mr_bind` bound to the local counter with `FI_REMOTE_WRITE` and  that was causing double increments (one from the local counter increment and another from the remote counter increment).